sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -95,12 +95,15 @@ abstract class Expression extends TreeNode[Expression] {
     ctx.subExprEliminationExprs.get(this).map { subExprState =>
       // This expression is repeated meaning the code to evaluated has already been added
       // as a function, `subExprState.fnName`. Just call that.
+      val isNull = ctx.freshName("isNull")
+      val primitive = ctx.freshName("primitive")
```

Comment:
Actually, the above variables are not unnecessary. I will remove them later if the test is passed.